lambda/shared/src/main/scala/feral/lambda/IOLambda.scala

```diff
@@ -18,14 +18,29 @@ package feral
 package lambda
 
 import cats.effect.IO
+import cats.effect.kernel.Resource
 import io.circe.Decoder
 import io.circe.Encoder
+import cats.effect.IOLocal
 
 abstract class IOLambda[Event, Result](
     implicit private[lambda] val decoder: Decoder[Event],
     private[lambda] val encoder: Encoder[Result]
 ) extends IOLambdaPlatform[Event, Result]
     with IOSetup {
 
-  def apply(event: Event, context: Context, setup: Setup): IO[Option[Result]]
+  final type Setup = (Event, Context[IO]) => IO[Option[Result]]
+  final override protected def setup: Resource[IO, Setup] =
+    Resource
+      .eval((IOLocal(null.asInstanceOf[Event]).product(IOLocal[Context[IO]](null))))
+      .flatMap {
+        case (localEvent, localContext) =>
+          handler(LambdaEnv.ioLambdaEnv(localEvent, localContext)).map {
+            result => (event, context) =>
+              localEvent.set(event) *> localContext.set(context) *> result
+          }
+      }
+
+  def handler(implicit env: LambdaEnv[IO, Event]): Resource[IO, IO[Option[Result]]]
```

Comment:
I think this is the most confusing aspect of this change: The `Resource` is run once, to create an `IO` that will be run many times. The `IO` is the handler.

Reply:
Hmm, this is unsafe, because it would let you access the `event` and `context` while building the resource stack, which is not right. I knew those `null`s were a bad sign 😅 . Fixing.